packages/cog/src/cog/cog.cutline.ts

```diff
@@ -0,0 +1,265 @@
+                for (let i = 0; i < lines.length; ++i) {
+                    const entry = lines[i];
+                    if (Array.isArray(entry)) {
+                        for (const l of entry) {
+                            if (lineIntersect(l, tiffLine).features.length > 0) foundp.push(polygons[i]);
+                            break;
+                        }
+                    } else if (lineIntersect(entry, tiffLine).features.length > 0) foundp.push(polygons[i]);
+                }
+                for (const p of polygons) {
+                    if (booleanWithin(p, tiffPoly)) foundp.push(p);
+                }
+                if (foundp.length != 0) cutTifs.set(path, await cutCallback(path, makeCutline(cutline, foundp)));
+            }),
+        );
+    }
+
+    await Promise.all(ps);
+
+    for (const b of bands) {
+        if (b instanceof VNodeElement) {
+            const children: VNode[] = [];
+            for (const elm of b.children) {
+                if (elm instanceof VNodeElement && (elm.tag === 'SimpleSource' || elm.tag === 'ComplexSource')) {
+                    const sf = elm.tags('SourceFilename').next();
+                    if (sf.done) continue;
+                    const path = sf.value.textContent;
+                    const cutPath = cutTifs.get(path);
+                    if (cutPath != null) {
+                        sf.value.textContent = cutPath;
+                        if (b.attrs.band === '4' && elm.tag === 'ComplexSource') {
+                            elm.children = elm.children.filter(
+                                (e) => !(e instanceof VNodeElement && e.tag.startsWith('Scale')),
+                            );
+                            for (const e of elm.tags('SourceBand')) e.textContent = '4';
+                        }
+                    } else if (!stetTifs.has(path)) {
+                        continue;
+                    }
+                }
+                children.push(elm);
+            }
+            b.children = children;
+        }
+    }
+
+    logger.info({ total: cutTifs.size + stetTifs.size, cropped: cutTifs.size }, 'Source Tiff count');
+
+    return vrt;
+}
+
+/**
+ * Run gdalwarp on a tiff with a cutline
+ *
+ * @param source the path to the source tiff
+ * @param target the path to the tiff to create
+ * @param cutline defines the cutline
+ * @param cutlinePath the path to the cutline to create
+ * @param cutlineBlend a number > 0 to blend the cutline
+ * @param outputFs the file processor used to create cutline and target
+ * @param logger
+ */
+export async function cutTiff(
+    source: string,
+    target: string,
+    cutline: FeatureCollection,
+    cutlinePath: string,
+    cutlineBlend: number,
+    outputFs: FileProcessor,
+    logger: LogType,
+): Promise<void> {
+    const gdal = Env.get(Env.Gdal.UseDocker, undefined) ? new GdalDocker() : new GdalLocal();
+
+    gdal.mount?.(source);
+    gdal.mount?.(target);
+    gdal.mount?.(cutlinePath);
```

Comment:
do we need to mount all of these,

can we just mount the temp folder that is created?